quesma/queryparser/pancake_sql_query_generation.go

```diff
@@ -191,7 +192,39 @@ func (p *pancakeSqlQueryGenerator) generateBucketSqlParts(bucketAggregation *pan
 	return
 }
 
-// TODO: deduplicate metric names
+func (p *pancakeSqlQueryGenerator) generateLeafFilter(layer *pancakeModelLayer, whereClause model.Expr) (addSelectColumns []model.AliasedExpr, err error) {
+	if layer == nil { // no metric aggregations in filter
+		return nil, nil
+	}
+	if layer.nextBucketAggregation != nil {
+		return nil, errors.New("filter layer can't have sub bucket aggregations")
+	}
+
+	for _, metric := range layer.currentMetricAggregations {
+		for columnId, column := range metric.selectedColumns {
+			aliasedName := fmt.Sprintf("%s_col_%d", metric.internalName, columnId)
+			// Add if
+			var columnWithIf model.Expr
+			switch function := column.(type) {
+			case model.FunctionExpr:
+				if function.Name == "count" {
+					columnWithIf = model.NewFunction("countIf", whereClause)
+				} else if len(function.Args) == 1 {
+					columnWithIf = model.NewFunction(function.Name+"If", function.Args[0], whereClause)
```

Comment:
Is this correct to do that (add `if`) for any function with one argument?

Reply:
Seems so:
https://clickhouse.com/docs/en/sql-reference/aggregate-functions/combinators#-if

Reply:
I would add this link as a comment about rationale

Reply:
We can have a 'builder' function here.